lib/puppet/provider/mount.rb

```diff
@@ -44,6 +47,9 @@ def remount
   def unmount
     umount(resource[:name])
 
+    # Remove the mountpoint directory if it exists and manage_mountpoint is true
+    Dir.rmdir(resource[:name]) if resource[:manage_mountpoint] && File.exist?(resource[:name])
```

Comment:
I don't think we should even be attempting to remove the directory on unmount. If the user really don't want this to exist anymore, they should explicitly set that in a file resource. Otherwise this is inconsistent with established behavior. As far as I'm aware, puppet doesn't clean anything up unless explicitely told to.

I would also suggest changing the param name to something like `create_mountpoint` to more accurately reflect this change.